runelite-client/src/main/java/net/runelite/client/plugins/mta/telekinetic/TelekineticRoom.java

```diff
@@ -71,12 +71,15 @@
 
 	private final Client client;
 
+	private Set<WallObject> telekineticWalls = new HashSet<>();

```

Comment:
this should be final, also, probs just arraylist is enough

Reply:
also clear it on shutdown

Reply:
I don't really agree, a set seems better here imo. Changed anyway, but seems worse for performance